homeassistant/components/lamarzocco/services.py

```diff
@@ -0,0 +1,163 @@
+
+from .const import (
+    CONF_CONFIG_ENTRY,
+    CONF_DAY_OF_WEEK,
+    CONF_ENABLE,
+    CONF_HOUR_OFF,
+    CONF_HOUR_ON,
+    CONF_MINUTE_OFF,
+    CONF_MINUTE_ON,
+    DAYS,
+    DOMAIN,
+    SERVICE_AUTO_ON_OFF_ENABLE,
+    SERVICE_AUTO_ON_OFF_TIMES,
+)
+from .coordinator import LaMarzoccoUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+CONFIG_ENTRY_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CONFIG_ENTRY): selector.ConfigEntrySelector(
+            {
+                "integration": DOMAIN,
+            }
+        )
+    }
+)
+
+
+SET_AUTO_ON_OFF_ENABLE_SCHEMA = CONFIG_ENTRY_SCHEMA.extend(
+    {
+        vol.Required(CONF_DAY_OF_WEEK): vol.In(DAYS),
+        vol.Required(CONF_ENABLE): cv.boolean,
+    }
+)
+
+SET_AUTO_ON_OFF_TIMES_SCHEMA = CONFIG_ENTRY_SCHEMA.extend(
+    {
+        vol.Required(CONF_DAY_OF_WEEK): vol.In(DAYS),
+        vol.Required(CONF_HOUR_ON): vol.All(vol.Coerce(int), vol.Range(min=0, max=23)),
+        vol.Optional(CONF_MINUTE_ON, default=0): vol.All(
+            vol.Coerce(int), vol.Range(min=0, max=59)
+        ),
+        vol.Required(CONF_HOUR_OFF): vol.All(vol.Coerce(int), vol.Range(min=0, max=23)),
+        vol.Optional(CONF_MINUTE_OFF, default=0): vol.All(
+            vol.Coerce(int), vol.Range(min=0, max=59)
+        ),
+    }
+)
+
+
+async def __call_service(
+    func: Callable[..., Coroutine[Any, Any, Any]], *args: Any, **kwargs: Any
+) -> None:
+    """Call a service and handle exceptions."""
+    try:
+        await func(*args, **kwargs)
+    except (AuthFail, RequestNotSuccessful, TimeoutError) as ex:
+        raise HomeAssistantError("Service call encountered error: %s" % str(ex)) from ex
+
+
+def __get_coordinator(
+    hass: HomeAssistant, call: ServiceCall
+) -> LaMarzoccoUpdateCoordinator:
+    """Get the coordinator from the entry."""
+    entry_id: str = call.data[CONF_CONFIG_ENTRY]
+    entry: ConfigEntry | None = hass.config_entries.async_get_entry(entry_id)
+
+    if not entry:
+        raise ServiceValidationError(
+            f"Invalid config entry: {entry_id}",
+            translation_domain=DOMAIN,
+            translation_key="invalid_config_entry",
+            translation_placeholders={
+                "config_entry": entry_id,
+            },
+        )
+    if entry.state != ConfigEntryState.LOADED:
+        raise ServiceValidationError(
+            f"Config entry {entry.title} is not loaded",
+            translation_domain=DOMAIN,
+            translation_key="unloaded_config_entry",
+            translation_placeholders={
+                "config_entry": entry.title,
+            },
+        )
+    coordinator: LaMarzoccoUpdateCoordinator = hass.data[DOMAIN][entry_id]
+    return coordinator
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Create and register services for the La Marzocco integration."""
+
+    async def _set_auto_on_off_enable(service: ServiceCall) -> None:
```

Comment:
Why do we need two services for one use case?

Reply:
because that is how it is in the app. You can enable/disable the schedule for a specific day, but independently set the schedule for each day. If I have it in one service you would always need to set the schedule if you want to en-/disable one day.

Reply:
and because I find it weird if you had a either put enabled, or all the other attributes, which then would be custom validation logic (afaik) which I find annoying from a UX perspective.